### PR TITLE
*: cut v1.7.0-rc.0

### PR DESCRIPTION
**Pull request by brancz**
We've collected quite a few things and I don't think there are any features outstanding that we want to get into 1.7, so here it goes.

@mxinden @andyxning @tariq1890 @LiliC 

cc @serathius @piosz @DirectXMan12 please stand by to publish the gcr.io image :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+* [CHANGE] use appsv1 apigroup for ReplicaSet. #804
+* [CHANGE] Use distroless/static as base image to further reduce image size. #748
+* [CHANGE] Return standardized text for health endpoint. #767
+* [ENHANCEMENT] Add DeadlineExceeded pod termination reason. #740
+* [ENHANCEMENT] Add CreateContainerError and InvalidImageName waiting reasons. #763
+* [FEATURE] Add pod init container metrics. #762
+* [FEATURE] Add Metrics About StorageClass. #777
+* [FEATURE] Allow black/white- listing of metrics by regex. #773
+* [FEATURE] Add metrics for VerticalPodAutoscaler objects (experimental and disabled by default). #791
+* [FEATURE] Add (experimental) kube_*_annotations metrics. #770
```
Comment (LiliC) [inline CHANGELOG.md]: Nit: To stay consistent with the formatting style of the CHANGELOG:
```suggestion
* [FEATURE] Add (experimental) `kube_*_annotations metrics`. #770
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
+## v1.7.0-rc.0 / 2019-07-10
+
+* [CHANGE] use appsv1 apigroup for ReplicaSet. #804
```
Comment (LiliC) [inline CHANGELOG.md]:
```suggestion
* [CHANGE] Use appsv1 apigroup for ReplicaSet. #804
```

**Comment by tariq1890** [comment]
Let's update the version support matrix as well. We need to add a row for v1.7.0.  This needs to be recorded in Changelog as well.

**Comment by andyxning** [comment]
How about the image versions?

**Comment by brancz** [comment]
> Let's update the version support matrix in the Readme. We need to add a row for v1.7.0. This needs to be recorded in Changelog as well.

Will do!

> How about the image versions?

We historically never updated the manifests with pre-releases.

**Comment by brancz** [comment]
@tariq1890 done. Please take a look :slightly_smiling_face: .

**Comment by brancz** [comment]
Container image is published at `quay.io/coreos/kube-state-metrics:v1.7.0-rc.0`

**Pull request merged**